code/modules/mob/living/simple_animal/hostile/mining/elites/elite.dm

```diff
@@ -339,6 +347,30 @@ While using this makes the system rely on OnFire, it still gives options for tim
 		visible_message("<span class='warning'>[mychild] suddenly reappears above [src]!</span>")
 		playsound(loc,'sound/effects/phasein.ogg', 200, 0, 50, TRUE, TRUE)
 
+/obj/structure/elite_tumor/HasProximity(atom/movable/AM)
+	if(!invader_checking)
+		return
+	if(isliving(AM))
+		var/mob/living/M = AM
+		if(!ishuman(M) && !isrobot(M))
+			return
+		if(M == activator)
+			return
+		if(M in invaders)
+			to_chat(M, "<span class='colossus'><b>You dare to try to break the sanctity of our arena? SUFFER...</b></span>")
+			for(var/i in 1 to 4)
+				M.apply_status_effect(STATUS_EFFECT_VOID_PRICE) /// Hey kids, want 60 brute damage, increased by 40 each time you do it? Well, here you go!
+		else
+			to_chat(M, "<span class='userdanger'>Only spectators are allowed, while the arena is in combat...</span>")
+			invaders += M
+		var/list/valid_turfs = list()
+		for(var/turf/T in RANGE_TURFS(13, src))
```

Comment:
I'm unsure on what this proc does. Will it check every single turf within 13 tiles of the tumor?

Reply:
`RANGE_TURFS` will return all the turfs in a radius as if you call it using `range` instead of `view`. It is done using `block` which I guess is faster than `range`.

I've tested this new define and profiled it:

```dm
// Gets the top, bottom, left and right edges
#define RANGE_EDGE_TURFS(RADIUS, CENTER) \
  block(locate(max(CENTER.x - RADIUS, 1), max(CENTER.y - RADIUS, 1), CENTER.z), locate(max(CENTER.x + RADIUS, 1), max(CENTER.y - RADIUS, 1), CENTER.z)) +\
  block(locate(max(CENTER.x - RADIUS, 1), max(CENTER.y + RADIUS, 1), CENTER.z), locate(max(CENTER.x + RADIUS, 1), max(CENTER.y + RADIUS, 1), CENTER.z)) +\
  block(locate(max(CENTER.x - RADIUS, 1), max(CENTER.y - RADIUS + 1, 1), CENTER.z), locate(max(CENTER.x - RADIUS, 1), max(CENTER.y + RADIUS - 1, 1), CENTER.z)) +\
  block(locate(max(CENTER.x + RADIUS, 1), max(CENTER.y - RADIUS + 1, 1), CENTER.z), locate(max(CENTER.x + RADIUS, 1), max(CENTER.y + RADIUS - 1, 1), CENTER.z))
```
Basically just gets the edges of a radius around the center.

Benchmark code and results:
```dm
/mob/proc/test(radius, n)
	var/turf/C = get_turf(src)
	edge_test(radius, C, n)
	range_turf_test(radius, C, n)

/mob/proc/range_turf_test(radius, turf/center, n)
	for(var/i in 1 to n)
		var/list/result = list()
		for(var/turf/T in RANGE_TURFS(radius, center))
			if(get_dist(T, center) == radius)
				result += T

/mob/proc/edge_test(radius, turf/center, n)
	for(var/i in 1 to n)
		var/list/result = RANGE_EDGE_TURFS(radius, center)
```

```
radius = 1
Proc Name                                                                                Self CPU    Total CPU    Real Time     Overtime        Calls
------------------------------------------------------------------------------------    ---------    ---------    ---------    ---------    ---------
/mob/proc/test                                                                              0.000        0.589        0.589        0.000            1
/mob/proc/edge_test                                                                         0.326        0.326        0.326        0.326            1
/mob/proc/range_turf_test                                                                   0.263        0.263        0.263        0.263            1

radius = 2
Proc Name                                                                                Self CPU    Total CPU    Real Time     Overtime        Calls
------------------------------------------------------------------------------------    ---------    ---------    ---------    ---------    ---------
/mob/proc/test                                                                              0.000        1.010        1.025        0.000            1
/mob/proc/range_turf_test                                                                   0.665        0.665        0.681        0.665            1
/mob/proc/edge_test                                                                         0.345        0.345        0.344        0.345            1

radius = 5
Proc Name                                                                                Self CPU    Total CPU    Real Time     Overtime        Calls
------------------------------------------------------------------------------------    ---------    ---------    ---------    ---------    ---------
/mob/proc/test                                                                              0.000        3.225        3.279        0.000            1
/mob/proc/range_turf_test                                                                   2.825        2.825        2.879        2.775            1
/mob/proc/edge_test                                                                         0.400        0.400        0.400        0.400            1

radius = 10
Proc Name                                                                                Self CPU    Total CPU    Real Time     Overtime        Calls
------------------------------------------------------------------------------------    ---------    ---------    ---------    ---------    ---------
/mob/proc/test                                                                              0.000        8.292        8.482        0.000            1
/mob/proc/range_turf_test                                                                   7.793        7.793        7.984        7.793            1
/mob/proc/edge_test                                                                         0.499        0.499        0.498        0.499            1
```

As you can see it performs better in all situations except with a radius of 1. Also easier to use.

Reply:
implimented a version with henk